### PR TITLE
[Docs] Fix Data examples TOC and rendering

### PR DESCRIPTION
**Pull request by amogkam**
https://github.com/ray-project/ray/pull/36206/commits/b10f19dd38533cc4caef4e3391dcf5d0db939c37 renamed the examples file for Ray Data.

We have to make the corresponding change to the TOC, so that the page still shows up on the docs.

Also removes the example files from `exclude_patterns` so they are built correctly.
Includes the examples in a hidden toctree to prevent sphinx warnings.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by matthewdeng — APPROVED** [review]

File: doc/source/_toc.yml [hunk 1]
```diff
@@ -96,7 +96,7 @@ parts:
           - file: data/getting-started
           - file: data/key-concepts
           - file: data/user-guide
-          - file: data/examples
+          - file: data/data_examples
```
Comment (angelinalg) [inline doc/source/_toc.yml:99]:
```suggestion
          - file: data/data_examples
            title: Examples
```

Comment (angelinalg) [reply doc/source/_toc.yml:99]: Are you open to keeping the name the same?

**Review by angelinalg — APPROVED** [review]

File: doc/source/_toc.yml (hunk 1 above)
Comment (amogkam) [reply doc/source/_toc.yml:99]: The name isn't changed in this PR. It's still being pulled from the title of data_examples.rst

Comment (angelinalg) [reply doc/source/_toc.yml:99]: I think you can set it with that `title` line, no?

Comment (amogkam) [reply doc/source/_toc.yml:99]: If not provided, it will default to the title in the rst file

**Review by maxpumperla — APPROVED** [review]
Thanks! this is obviously much better.

File: doc/source/data/examples/index.rst [hunk 2]
```diff
@@ -4,6 +4,23 @@
...
+    :hidden:
+
+    huggingface_vit_batch_prediction
+    pytorch_resnet_batch_prediction
+    batch_inference_object_detection
+    nyc_taxi_basic_processing
+    batch_training
+    ocr_example
+    random-access
+    custom-datasource
```
Comment (bveeramani) [inline doc/source/data/examples/index.rst:20]: These will show up in the navbar.

**Review by bveeramani — APPROVED** [review]

File: doc/source/data/examples/index.rst (hunk 2 above)
Comment (amogkam) [reply doc/source/data/examples/index.rst:20]: yep, as discussed offline, doesn't seem there's an easy way to avoid this

**Pull request merged**